implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -255,11 +249,16 @@ private DimensionList fromTags(List<Tag> tags) {
 
     private void send(List<String> metricLines) {
         try {
+            String endpoint = config.uri();
+            reportErrorIfEndpointSeemsInvalid(endpoint);
+
+            logger.debug("Sending {} lines to {}", metricLines.size(), endpoint);
+
             String body = String.join("\n", metricLines);
             logger.debug("Sending lines:\n{}", body);
 
-            HttpSender.Request.Builder requestBuilder = httpClient.post(endpoint);
-            if (!ignoreToken) {
+            HttpSender.Request.Builder requestBuilder = httpClient.post(config.uri());
```

Comment:
Will httpClient handle invalid URIs gracefully?

Reply:
```suggestion
            HttpSender.Request.Builder requestBuilder = httpClient.post(endpoint);
```
This ensure that the same value that was checked and logged above is used in the rare case that it would change in between those few lines.

Reply:
I double checked, and the HttpClient does the same validation as we do in the check if the endpoint is valid. Therefore, I changed the method to check if the endpoint is valid and the `send` method. Now, an attempt to send metrics is only done if the endpoint is valid.